### ISSUE TITLE
SFT on math data cannot get the results in the paper

### ISSUE THREAD

**Issue by YJiangcm** [Timothy023/RLMEC#4]
Thank you for your excellent work!

I met a problem when reproducing **SFT LLaMA2-7B on MathInst** training data provided in this repository. I used the same training parameters as those in the paper, and evaluated the well-trained model using your code. However, the performance I got was far away from that in the paper:

|  | GSM8k | MATH | SVAMP | MM | Avg. |
|:-----------------------:|:-----:|:----------:|:---------:|:-----:|:-----:|
SFT LLM in the paper | 51.02 | 10.48 | 47.80 | 38.50 | 36.95|
My SFT LLM | 35.63 | 6.52 | 42.10 | 31.11 | 28.84 |

I utilized [LLaMA-Factory](https://github.com/hiyouga/LLaMA-Factory) to reproduce the results on 4×A800 GPUs, and the parameters I used are as follows:
```
### method
stage: sft
do_train: true
finetuning_type: full

### ddp
ddp_timeout: 180000000
deepspeed: examples/deepspeed/ds_z3_config.json

### dataset
template: alpaca
cutoff_len: 1500

### train
per_device_train_batch_size: 2
gradient_accumulation_steps: 96
learning_rate: 5.0e-6
num_train_epochs: 1.0
lr_scheduler_type: cosine
warmup_ratio: 0.1
fp16: true
```

This is the training loss obtained:
![training_loss](https://github.com/Timothy023/RLMEC/assets/70904956/df44c0ef-0355-4d49-a214-a9f505212985)

By the way, I can reproduce the results of SFT LLM on QA tasks by using LLaMA-Factory:
|  | ECQA | QASC | OBQA | ARC | Avg. |
|:-----------------------:|:-----:|:----------:|:---------:|:-----:|:-----:|
SFT LLM in the paper | 71.88 | 55.40 | 52.00 | 56.27 | 58.89 |
My SFT LLM | 70.65 | 52.27 | 51.60 | 59.81 | 58.58 |

Therefore, I wonder if my SFT has any problems and sincerely hope you can give me some instructions.

Looking forward to your reply!

**Comment by Timothy023** [comment]
In our SFT process, we adopt the following hyper-parameters:

+ batch_size = 128
+ epoch = 3
+ learning rate = 2e-5
+ warmup_ratio = 0.03

Hope these details can help you!

**Comment by YJiangcm** [comment]
Thanks for your kind reply!

I fixed batch_size = 128 and warmup_ratio = 0.03, then tried a grid search of the following parameters: 
`epoch = {1,2,3}, learning rate = {1e-5, 2e-5, 3e-5}`

However, the best performance I obtained is still 2 points lower than the result in the paper (avg. 34.81 v.s. avg. 36.95). Do you have any insights? Thanks a lot!

**Comment by Timothy023** [comment]
We follow the code in [stanford_alpaca](https://github.com/tatsu-lab/stanford_alpaca) to perform the SFT. Besides, we do not use any tricks in the SFT process.
